tests/src/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -22,6 +99,63 @@ public class CoreclrTestWrapperLib
         public const int DEFAULT_TIMEOUT = 1000 * 60*10;
         public const string GC_STRESS_LEVEL = "__GCSTRESSLEVEL";
 
+        public const string COLLECT_DUMPS_ENVIRONMENT_VAR = "__CollectDumps";
+        public const string CRASH_DUMP_FOLDER_ENVIRONMENT_VAR = "__CrashDumpFolder";
+
+        static bool CollectCrashDump(Process process, string path)
+        {
+            using (var crashDump = File.OpenWrite(path))
+            {
+                var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
+                return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+            }
+        }
+
+        static unsafe bool TryFindChildProcessByName(Process process, string childName, out Process child)
+        {
+            IntPtr snapshot = Kernel32.CreateToolhelp32Snapshot(Kernel32.Toolhelp32Flags.TH32CS_SNAPPROCESS, 0);
+            if (snapshot == IntPtr.Zero)
+            {
+                child = null;
+                return false;
+            }
+
+            try
+            {
+                int ppid = process.Id;
+
+                var processEntry = new Kernel32.ProcessEntry32 { Size = sizeof(Kernel32.ProcessEntry32) };
+
+                bool success = Kernel32.Process32First(snapshot, ref processEntry);
+                while (success)
+                {
+                    if (processEntry.ParentProcessID == ppid)
+                    {
+                        try
+                        {
+                            Process c = Process.GetProcessById(processEntry.ProcessID);
```

Comment:
Ah, right.  Sorry.  Are you missing a Dispose in the caller, then?  Not sure if you subscribe to the philosophy that resource management is stupid on the exit path of an app.